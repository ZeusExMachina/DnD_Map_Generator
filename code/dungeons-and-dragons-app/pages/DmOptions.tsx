import {View} from "react-native";
import React from "react";
import LargeButton from "../components/LargeButton";
import {Typography} from "@material-ui/core";


const DmOptions = () => {
    return (
        <View>
            <Typography variant={"h2"}>
                Player Options
            </Typography>
            <LargeButton buttonString={"Load existing game"} buttonRoute={"/dm"} />
            <LargeButton buttonString={"Create new game"} buttonRoute={"/dm"} />
        </View>
    );
}
export default DmOptions
