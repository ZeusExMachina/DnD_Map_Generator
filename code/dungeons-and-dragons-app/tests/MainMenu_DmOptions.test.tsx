import React from 'react';
import renderer from 'react-test-renderer'

import DmOptions from "../pages/DmOptions";

/*DM Options Page Rendering*/
describe('<MainMenu_DmOptions />', () => {

    test('Dm options page renders correctly', () => {
        const renderedComponent = renderer.create(<DmOptions />).toJSON()
        expect(renderedComponent).toMatchSnapshot();
    });

});