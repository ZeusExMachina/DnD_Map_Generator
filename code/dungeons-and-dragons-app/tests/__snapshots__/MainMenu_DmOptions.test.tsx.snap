// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainMenu_DmOptions /> Dm options page renders correctly 1`] = `
<View>
  <div
    className="backgroundImage"
  >
    <div
      className="dmOptionBox"
    >
      <h2
        className="MuiTypography-root dmTitle MuiTypography-h2"
      >
        Dungeon Master Options
      </h2>
      <div
        className="dmOptionLeft"
      >
        <h4
          className="MuiTypography-root dmSubtitle MuiTypography-h4"
        >
          Load Existing Game
        </h4>
        <div
          className="MuiFormControl-root MuiTextField-root makeStyles-gameCodeInput-1 MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="outlined-required"
            id="outlined-required-label"
            style={
              Object {
                "color": "#403940",
              }
            }
          >
            Enter a Game Code Here
            <span
              aria-hidden={true}
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
               
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <input
              aria-invalid={false}
              autoFocus={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              defaultValue=""
              disabled={false}
              id="outlined-required"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
            />
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-2 MuiOutlinedInput-notchedOutline"
            >
              <legend
                className="PrivateNotchedOutline-legendLabelled-4"
              >
                <span>
                  Enter a Game Code Here
                   *
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <div
          className="loadGameButtons"
        >
          <div
            className="SmallMenuButton"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-smallBackButton-6 MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiButton-startIcon MuiButton-iconSizeSmall"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11.67 3.87L9.9 2.1 0 12l9.9 9.9 1.77-1.77L3.54 12z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTypography-root makeStyles-buttonText-7 MuiTypography-button"
                >
                  Back
                </span>
              </span>
            </button>
          </div>
          <div
            className="SmallMenuButton"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-smallMenuButton-8 MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiTypography-root makeStyles-buttonText-9 MuiTypography-button"
                >
                  Load Game
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
      <div
        className="dmOptionRight"
      >
        <h4
          className="MuiTypography-root dmSubtitle MuiTypography-h4"
        >
          New Game
        </h4>
        <div
          className="themeSelect"
        >
          <div
            className="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
          >
            <ul
              className="MuiList-root MuiList-padding"
              onKeyDown={[Function]}
              role="menu"
              tabIndex={-1}
            >
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={0}
              >
                Caves
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Dungeon
              </li>
              <li
                aria-disabled={false}
                className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                Underground Mansion
              </li>
            </ul>
          </div>
        </div>
        <div
          className="createButton"
        >
          <div
            className="SmallMenuButton"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-smallMenuButton-8 MuiButton-outlinedPrimary MuiButton-outlinedSizeSmall MuiButton-sizeSmall"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                <span
                  className="MuiTypography-root makeStyles-buttonText-9 MuiTypography-button"
                >
                  Create
                </span>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</View>
`;
